### PR TITLE
refactor: ingest data

### PR DESCRIPTION
**Pull request by nicecui**
## What's Changed in this PR

as title

## Checklist

- [x] Please confirm that all corresponding versions of the documents have been revised.
- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/nightly/en/user-guide/ingest-data/for-iot/influxdb-line-protocol.md [hunk 1]
```diff
@@ -1,11 +1,7 @@
-# InfluxDB Line Protocol
+# Influxdb Line Protocol
```
Comment (sunng87) [inline docs/nightly/en/user-guide/ingest-data/for-iot/influxdb-line-protocol.md:1]: we can keep InfluxDB for consistency

File: docs/nightly/en/user-guide/ingest-data/for-iot/sql.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+# SQL
+
+We will use the `monitor` table as an example to show how to write data.
+
```
Comment (sunng87) [inline docs/nightly/en/user-guide/ingest-data/for-iot/sql.md:4]: We need to mention MySQL and PostgreSQL compatibility here. For example, how to use those database drivers to connect to the database

File: docs/nightly/en/user-guide/ingest-data/for-observerbility/opentelemetry.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# OpenTelemetry Protocol (OTLP)
+
+[OpenTelemetry](https://opentelemetry.io/) is a vendor-neutral open-source observability framework for instrumenting, generating, collecting, and exporting telemetry data such as traces, metrics, logs. The OpenTelemetry Protocol (OTLP) defines the encoding, transport, and delivery mechanism of telemetry data between telemetry sources, intermediate processes such as collectors and telemetry backends.
+
```
Comment (sunng87) [inline docs/nightly/en/user-guide/ingest-data/for-observerbility/opentelemetry.md:4]: OTLP is the underlying protocol. We also need to cover how to configure tools from OpenTelemetry ecosystem to ingest data. But this can be authored in another pr.

File: docs/nightly/en/user-guide/ingest-data/for-observerbility/overview.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Overview
+
+In observability scenarios,
+the ability to monitor and analyze system performance in real-time is crucial.
+GreptimeDB integrates seamlessly with leading observability tools to provide a comprehensive view of your system's health and performance metrics. 
+
+- [Prometheus](prometheus.md)
+- [Vector](vector.md)
+- [OpenTelemetry](opentelemetry.md)
```
Comment (sunng87) [inline docs/nightly/en/user-guide/ingest-data/for-observerbility/overview.md:9]: Telegraf

File: docs/nightly/en/user-guide/ingest-data/for-iot/influxdb-line-protocol.md (hunk 1 above)
Comment (nicecui) [reply docs/nightly/en/user-guide/ingest-data/for-iot/influxdb-line-protocol.md:1]: hmmm...? there is nothing changed here, just modified the typo

File: docs/nightly/en/user-guide/concepts/data-model.md [hunk 5]
```diff
@@ -95,4 +95,4 @@ GreptimeDB is designed on top of Table for the following reasons:
   The multi-value model is used to model data sources, where a metric can have multiple values represented by fields.
   The advantage of the multi-value model is that it can write or read multiple values to the database at once, reducing transfer traffic and simplifying queries. In contrast, the single-value model requires splitting the data into multiple records. Read the [blog](https://greptime.com/blogs/2024-05-09-prometheus) for more detailed benefits of multi-value mode.
 
-GreptimeDB uses SQL to manage table schema. Please refer to [table management](../table-management.md) for more information. However, our definition of schema is not mandatory and leans towards a **schemaless** approach, similar to MongoDB. For more details, see [Automatic Schema Generation](../write-data/overview.md#automatic-schema-generation).
+GreptimeDB uses SQL to manage table schema. Please refer to [table management](../table-management.md) for more information. However, our definition of schema is not mandatory and leans towards a **schemaless** approach, similar to MongoDB. For more details, see [Automatic Schema Generation](../ingest-data/overview.md#automatic-schema-generation).
```
Comment (killme2008) [inline docs/nightly/en/user-guide/concepts/data-model.md:98]: Always use the `/user-guide/table-management.md` and `/user-guide/ingest-data/overview...` is a good practice. The relative path is confused sometimes.

File: docs/nightly/en/user-guide/ingest-data/for-iot/grpc-sdks/template.md [hunk 6]
```diff
@@ -1,39 +1,34 @@
 
-GreptimeDB uses different client libraries for writing and querying data.
-You can choose the client library that best suits your needs.
-
-## Write data
-
-GreptimeDB provides an ingester library to help you write data.
+GreptimeDB provides an ingester libraries to help you write data.
```
Comment (killme2008) [inline docs/nightly/en/user-guide/ingest-data/for-iot/grpc-sdks/template.md:2]:
```suggestion
GreptimeDB offers ingester libraries for high-throughput data writing.
```

File: docs/nightly/en/user-guide/ingest-data/for-iot/grpc-sdks/template.md [hunk 7]
```diff
@@ -51,57 +46,55 @@ To improve the efficiency of writing data, you can create multiple rows at once
...
 
 {template ingester-lib-debug-logs%%}
 
-### Ingester library reference
+## Ingester library reference
 
 {template ingester-lib-reference%%}
 
-## Query data
+<!-- ## Query data
```
Comment (killme2008) [inline docs/nightly/en/user-guide/ingest-data/for-iot/grpc-sdks/template.md:97]: Adding a link to `Query data` document would be good.

File: docs/nightly/en/user-guide/ingest-data/for-iot/sql.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
+# SQL
+
+You can execute SQL statements using [MySQL](/user-guide/clients/mysql.md) or [PostgreSQL](/user-guide/clients/postgresql.md) clients.
```
Comment (killme2008) [inline docs/nightly/en/user-guide/ingest-data/for-iot/sql.md:3]:
```suggestion
You can execute SQL statements using [MySQL](/user-guide/clients/mysql.md) or [PostgreSQL](/user-guide/clients/postgresql.md) clients. You can access GreptimeDB using the MySQL or PostgreSQL protocol with any programming language you prefer, such as Java JDBC.
```

File: docs/nightly/en/user-guide/ingest-data/for-iot/sql.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+The result in the `UTC` time zone client is as follows:
+
+```sql
++-----------+---------------------+------+--------+
+| host      | ts                  | cpu  | memory |
++-----------+---------------------+------+--------+
+| 127.0.0.1 | 2023-12-31 16:00:00 |  0.4 |    0.1 |
+| 127.0.0.2 | 2023-12-31 16:00:00 |  0.5 |    0.1 |
++-----------+---------------------+------+--------+
+```
```
Comment (killme2008) [inline docs/nightly/en/user-guide/ingest-data/for-iot/sql.md:115]: It would be great to link the time zone settings in go or java programming language.

File: docs/nightly/en/user-guide/ingest-data/for-observerbility/prometheus.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+#    password: greptime_pwd
+```
+
+- The host and port in the URL represent the GreptimeDB server. In this example, the server is running on `localhost:4000`. You can replace it with your own server address. For the HTTP protocol configuration in GreptimeDB, please refer to the [protocol options](/user-guide/operations/configuration.md#protocol-options).
+- The `db` parameter in the URL represents the database to which we want to write data. It is optional. By default, the database is set to `public`.
+- `basic_auth` is the authentication configuration. Fill in the username and password if GreptimeDB authentication is enabled. Please refer to the [authentication document](/user-guide/clients/authentication.md).
+
+## Data Model
+
+In the [data model](/user-guide/concepts/data-model.md) of GreptimeDB, data is organized into tables with columns for tags, time indices, and fields.
```
Comment (killme2008) [inline docs/nightly/en/user-guide/ingest-data/for-observerbility/prometheus.md:33]:
```suggestion
In the [data model](/user-guide/concepts/data-model.md) of GreptimeDB, data is organized into tables with columns for tags, time index, and fields.
```

File: docs/nightly/en/user-guide/ingest-data/for-observerbility/prometheus.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+| greptime_value     | 500                                         | Double             |
+| greptime_timestamp | The sample's unix timestamp                 | Timestamp          |
+
+
+## Improve efficiency by using metric engine
+
+The Prometheus remote writing always creates a large number of small tables.
+These tables are classified as logical tables in GreptimeDB.
+However, having a large number of small tables can be inefficient for both data storage and query performance.
+To address this, GreptimeDB introduces the metric engine feature,
```
Comment (killme2008) [inline docs/nightly/en/user-guide/ingest-data/for-observerbility/prometheus.md:74]: Linked to contributor guide of metric engine.

File: docs/nightly/en/user-guide/ingest-data/for-observerbility/prometheus.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+| Tables                                                        |
++---------------------------------------------------------------+
+| prometheus_remote_storage_enqueue_retries_total               |
+| prometheus_remote_storage_exemplars_pending                   |
+| prometheus_remote_storage_read_request_duration_seconds_count |
+| prometheus_rule_group_duration_seconds                        |
+| ......                                                        |
++---------------------------------------------------------------+
+```
+
```
Comment (killme2008) [inline docs/nightly/en/user-guide/ingest-data/for-observerbility/prometheus.md:109]: Adding a document to describe physical table:

```
Of course, the physical table itself can also be queried. It contains columns from all the logical tables, making it convenient for multi-join analysis and computation.
```

File: docs/nightly/en/user-guide/ingest-data/overview.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+- [Vector](./for-observerbility/vector.md): Use GreptimeDB as a sink for Vector, ideal for complex data pipelines and diverse data sources.
+- [OpenTelemetry](./for-observerbility/opentelemetry.md): Collect and export telemetry data to GreptimeDB for detailed observability insights.
+- [Telegraf](./for-observerbility/telegraf.md): Ingest data from various sources to GreptimeDB using Telegraf, suitable for monitoring and performance analysis.
+
+### IoT Scenarios
+
+- [SQL INSERT](./for-iot/sql.md): A straightforward method for inserting data.
+- [gRPC SDK](./for-iot/grpc-sdks/overview.md): Offers efficient, high-performance data ingestion, particularly suited for real-time data applications and complex IoT infrastructures.
+- [InfluxDB Line Protocol](./for-iot/influxdb-line-protocol.md): A widely-used protocol for time-series data, facilitating migration from InfluxDB to GreptimeDB.
+- [EMQX](./for-iot/emqx.md): An MQTT broker supporting massive device connections, can be used to ingest data to GreptimeDB.
```
Comment (killme2008) [inline docs/nightly/en/user-guide/ingest-data/overview.md:26]:
```suggestion
- [EMQX](./for-iot/emqx.md): An MQTT broker supporting massive device connections, can ingest data to GreptimeDB directly.
```

File: docs/nightly/en/user-guide/manage-data/overview.md [hunk 14]
```diff
@@ -0,0 +1,196 @@
...
+```sql
++-----------+---------------------+------+--------+
+| host      | ts                  | cpu  | memory |
++-----------+---------------------+------+--------+
+| 127.0.0.1 | 2024-07-11 20:00:00 | 0.5  | 0.1    |
++-----------+---------------------+------+--------+
+1 row in set (0.01 sec)
+```
+
+Note that you **cannot omit** the other columns in the `INSERT INTO` statement if you only want to update one column.
```
Comment (killme2008) [inline docs/nightly/en/user-guide/manage-data/overview.md:76]: Since v0.9.0, we have provided a new option `merge_mode` when creating tables.

If choosing `last_none_null`, you don't have to present all the columns when updating.

https://docs.greptime.com/reference/sql/create#create-a-table-with-merge-mode

File: docs/nightly/en/user-guide/manage-data/overview.md [hunk 15]
```diff
@@ -0,0 +1,196 @@
...
+DELETE FROM monitor WHERE host='127.0.0.2' AND ts=1667446798450;
+```
+
+The output will be:
+
+```sql
+Query OK, 1 row affected (0.00 sec)
+```
+
+For more information about the `DELETE` statement, please refer to the [SQL DELETE](/reference/sql/delete.md).
```
Comment (killme2008) [inline docs/nightly/en/user-guide/manage-data/overview.md:196]: Add a new section to describe `truncata table`, which delete all table data in an efficient way.

For example `truncate table monitor;`

I'll add the `truncate` reference document.

File: docs/nightly/en/user-guide/manage-data/overview.md (hunk 14 above)
Comment (killme2008) [reply docs/nightly/en/user-guide/manage-data/overview.md:76]: Another note: the influxdb tables created automatically are both `append_only=true` and `merge_mode=last_none_null`, which is compatible with the influxdb protocol semantic.

**Review by killme2008 — COMMENTED** [review]
I just reviewd the en version, please take a look. And don't forget to update the zh docs. @nicecui

File: docs/nightly/en/greptimecloud/integrations/mysql.md [hunk 16]
```diff
@@ -36,7 +36,7 @@ Use following connect string for your JDBC client.
 jdbc:mysql://<host>:4002/<dbname>?user=<username>&password=<password>
 ```
 
-Read the [time zone setting](https://docs.greptime.com/user-guide/client-libraries/java#time-zone) for JDBC.
+Read the [time zone setting](https://docs.greptime.com/user-guide/ingest-data/for-iot/grpc-sdks/java#time-zone) for JDBC.
```
[the remark this answers is not in the document]
Comment (nicecui) [reply docs/nightly/en/greptimecloud/integrations/mysql.md:39]: Since we moved client-libraries to ingest-data, there is no place for the MySQL JDBC section in the Java SDK document now, as the content is belong to the query scope.
I'll comment out this link, and add it back when complete the SQL tools document.

File: docs/nightly/en/user-guide/concepts/data-model.md (hunk 5 above)
Comment (nicecui) [reply docs/nightly/en/user-guide/concepts/data-model.md:98]: This PR will not check all path writing methods, which will be modified with future PRs.

File: docs/nightly/en/user-guide/ingest-data/for-iot/grpc-sdks/template.md (hunk 7 above)
Comment (nicecui) [reply docs/nightly/en/user-guide/ingest-data/for-iot/grpc-sdks/template.md:97]: This section aims to introduce how to use other SDKs to query data, it will be moved to the SDK tool document later in the other refactor PRs.

File: docs/nightly/en/user-guide/ingest-data/for-iot/sql.md (hunk 9 above)
Comment (nicecui) [reply docs/nightly/en/user-guide/ingest-data/for-iot/sql.md:115]: The SQL time zone configuration for the specific programming languages will move to the SQL tool document in another refactor PR.

File: docs/nightly/en/user-guide/manage-data/overview.md (hunk 14 above)
Comment (nicecui) [reply docs/nightly/en/user-guide/manage-data/overview.md:76]: how about gRPC, OTLP, and all the other protocols which can create table automatically?

File: docs/nightly/en/user-guide/ingest-data/for-iot/influxdb-line-protocol.md (hunk 1 above)
Comment (sunng87) [reply docs/nightly/en/user-guide/ingest-data/for-iot/influxdb-line-protocol.md:1]: the type case here , we should keep `InfluxDB` where DB is in uppercase

File: docs/nightly/en/user-guide/ingest-data/for-iot/grpc-sdks/template.md (hunk 7 above)
Comment (nicecui) [reply docs/nightly/en/user-guide/ingest-data/for-iot/grpc-sdks/template.md:97]: the query data links are at `/ingest-data/overview.md#next-steps`

File: docs/nightly/en/user-guide/ingest-data/for-observerbility/overview.md (hunk 4 above)
Comment (nicecui) [reply docs/nightly/en/user-guide/ingest-data/for-observerbility/overview.md]: the telegraf section has been moved to `/user-guide/ingest-data/for-iot/influxdb-line-protocol.md#telegraf`

**Review by killme2008 — APPROVED** [review]
LGTM

Thank you, a great work!

**Review by sunng87 — APPROVED** [review]
